pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py

```diff
@@ -46,6 +46,99 @@ def test_sync_publish_update_info(self):
         """
+        _, update_list = self._set_repo_and_get_repo_data()
+
+        # getting the update info from the fixtures repo
+        update_info_fixtures = get_xml_content_from_fixture(
+            fixture_path=RPM_WITH_MODULES_FEED_URL,
+            data_type='updateinfo',
+        )
+        self.assertEqual(
+            self._get_errata_rpm_mapping(update_list),
+            self._get_errata_rpm_mapping(update_info_fixtures),
+            'mismatch in the module packages.'
+        )
+
+    def test_collection_field(self):
+        """Test the collection field in the update info.
+
+        This test provides the following
+
+        1. Check whether all the modules in the published repo
+            contains a collection field.
+        2. Check whether the collection field has proper name. The collection name computation
+            is as below.
+
+        The collection name is created using the information from fixtures that is stored in
+        a set {<errata-id>:<module-name>}. First, the set information is used in computing
+        a set ``collections_from_fixtures`` that maps the repo_id to the collection-name.
+        The collection-name set is computed using the logic <repo-id>_<index>_<module-name>.
+        The module name is ``default`` and the index is``0`` for ursine RPMs. The set is
+        created using set-comprehension and x-path. After creating the set,
+        it appears as in the example below.
+
+        example :
+        collections_from_fixtures = {
+            'RHEA..1' : 'repo_id_1_duck',
+            'RHEA..2' : 'repo_id_2_duck',
+            'RHEA..3' : 'repo_id_0_default'
+        }
+
+        This set is compared against the collection-name from the published repo's ``updateinfo``.
+        """
+        repo, update_list = self._set_repo_and_get_repo_data()
+
+        # getting the update info from the fixtures repo
+        update_info_fixtures = get_xml_content_from_fixture(
+            fixture_path=RPM_WITH_MODULES_FEED_URL,
+            data_type='updateinfo',
+        )
+
+        # Errata ID to Collection name map in Updateinfo of published repo.
+        collection_update_list = {
+            update.find('./id').text: update.find('.//collection').attrib['short']
+            for update in update_list.findall('update')
+        }
+
+        collections_from_fixtures = {
+            update.find('id').text:
+            'default' if update.find('.//module') is None
+            else update.find('.//module').attrib['name']
+            for update in
+            update_info_fixtures.findall('.//update')
+        }
+
+        # ``indexes`` is used to increase the index of the module in the collections
+        indexes = {}
+        for key, val in collections_from_fixtures.items():
+            if val in indexes:
+                indexes[val] += 1
+            else:
+                indexes[val] = 1
+            collections_from_fixtures[key] = (
```

Comment:
I am not so sure we should mutate a dictionary while iterating it.

in this case `.items()` returns a generator, which will get its values from a mutated dictionary. a simple solution would be evaluating it at the beginning"by explicitly transforming in a `tuple` or `list` so future changes will not affect the object being iterated.

```py
for key, val in tuple(collections_from_fixtures.items()):
```